go/vaas/pkg/broadcast_channel/broadcast_channel.go

```diff
@@ -0,0 +1,82 @@
+package broadcast_channel
+
+import (
+	"context"
+)
+
+type IBroadcastChannel[T any] interface {
+	Serve(ctx context.Context)
+	Subscribe()
+	RemoveSubscription(subscription <-chan T)
+}
+
+type BroadcastChannel[T any] struct {
+	input            <-chan T
+	subscribers      []chan T
+	addSubscriber    chan chan T
+	removeSubscriber chan (<-chan T)
+}
+
+func New[T any](ctx context.Context, input <-chan T) *BroadcastChannel[T] {
+	b := BroadcastChannel[T]{
+		input:            input,
+		subscribers:      make([]chan T, 0),
+		addSubscriber:    make(chan chan T),
+		removeSubscriber: make(chan (<-chan T)),
+	}
+
+	go b.Serve(ctx)
+
+	return &b
+}
+
+func (b *BroadcastChannel[T]) Serve(ctx context.Context) {
```

Comment:
You are leaking channels here. By convention, the one who creates a channel is the only one writing to it and the one who closes a channel. I think you are never closing your channels for removed subscribers.